src/main/scala/org/constellation/trust/TrustManager.scala

```diff
@@ -46,14 +56,39 @@ class TrustManager[F[_]](nodeId: Id, clusterStorage: ClusterStorageAlgebra[F])(i
           trustNodes = TrustManager.calculateTrustNodes(trustNodesInternal.distinct, nodeId, scoringMap)
           //          _ <- logger.debug(s"TrustManager.calculateTrustNodes for trustNodes: ${trustNodes.toString()}")
 
-          idMappedScores: Map[Id, Double] = SelfAvoidingWalk
-            .runWalkFeedbackUpdateSingleNode(
-              selfNodeId,
-              trustNodes
-            )
-            .edges
-            .map(e => idxMap(e.dst) -> e.trust) //.flatMap(e => idxMap.get(e.dst).map(i => i -> e.trust) )//todo bug here
-            .toMap
+          sawAttempt = Try {
+            SelfAvoidingWalk
+              .runWalkFeedbackUpdateSingleNode(
+                selfNodeId,
+                trustNodes
+              )
+              .edges
+              .map(e => idxMap(e.dst) -> e.trust) //.flatMap(e => idxMap.get(e.dst).map(i => i -> e.trust) )//todo bug here
+              .toMap
+          }
+
+          // Temporary fallback
+          idMappedScores: Map[Id, Double] = sawAttempt match {
+            case Success(x) => x
+            case Failure(e) =>
+              logger.error(s"SAW Failure: $e")
+              val selfScores = scores.filter{x => x.id == nodeId}.head.view
+              val selfM = collection.mutable.HashMap[Id, Double]()
+              selfScores.foreach{case (k, v) => selfM(k) = v}
+              val addressToId = selfScores.keys.map{k => k.address -> k }.toMap
+              if (this.eigenTrust != null) {
+                val addressMap = this.eigenTrust.getTrustForAddresses.unsafeRunSync()
```

Comment:
Please avoid `unsafeRunSync`. An effectful code should be wrapped into a monad that will delay the execution of side effects. Running `unsafeRunSync` manually is both unsafe and performance degrading.

Reply:
Fixed, this code is hopefully only temporary until larger fix implemented -- and should be fast since it's only accessing a map, but I rewrote it as monadic anyways.